les/peer.go

```diff
@@ -68,8 +71,12 @@ const (
 )
 
 type peer struct {
-	*p2p.Peer
+	// WARNING: The `invalidReq` field is accessed atomically. On 32 bit platforms, only
+	// 64-bit aligned fields can be atomic. The struct is guaranteed to be so aligned,
+	// so take advantage of that (https://golang.org/pkg/sync/atomic/#pkg-note-BUG).
+	invalidReq uint64
```

Comment:
I think we can make this `uint32`.That way we can place it in the correct location within the struct fields and don't need this alignment hack. There's no legitimate need for the counter to be 64 bit wide.

Btw, do we need it to be atomic at all? I think we only ever touch it inside handleMsg, which executes serially. Maybe atomics are good for future proofing, just thinking out aloud here.

Reply:
Server handlers will process all requests async, so atomic operation here is necessary.
But we can use uint32 instead, thanks for pointing it out!